src/transactions/SponsorshipUtils.cpp

```diff
@@ -148,18 +148,22 @@ isSignerSponsored(std::vector<Signer>::const_iterator const& signerIt,
 static uint32_t
 computeMultiplier(LedgerEntry const& le)
 {
-    auto type = le.data.type();
-    if (type == ACCOUNT)
+    switch (le.data.type())
     {
+    case ACCOUNT:
         return 2;
-    }
-    else if (type == CLAIMABLE_BALANCE)
-    {
+    case TRUSTLINE:
+    case OFFER:
+    case DATA:
+        return 1;
+    case CLAIMABLE_BALANCE:
         return static_cast<uint32_t>(
             le.data.claimableBalance().claimants.size());
+    case LIQUIDITY_POOL:
```

Comment:
Hmm that's a good point. `computeMultiplier` and `isSubentry` is used for non-sponsored `LedgerEntry` creation, so I thought it would make sense to fill the `switch` statements out accurately for now. But you're right in that we don't actually need to use any of these methods. We just need to use `LedgerTxn::create`.

Reply:
Updated to throw